File: gaia-sdk-javascript/src/graphql/response/type/BehaviourConnection.ts

import {PageInfo} from "./PageInfo";
import {BehaviourEdge} from "./BehaviourEdge";

import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../../request/enumeration/RuntimeState";
import {SkillState} from "../../request/enumeration/SkillState";

/**
* Represents a paginated collection of behaviours
*/
export interface BehaviourConnection {
    edges?:[BehaviourEdge], 
    pageInfo?:PageInfo
}

File: gaia-sdk-python/gaia_sdk/graphql/response/type/BehaviourConnection.py

from gaia_sdk.graphql.response.type.PageInfo import PageInfo
from gaia_sdk.graphql.response.type.BehaviourEdge import BehaviourEdge

from dataclasses import dataclass
from typing import List
Uuid = str
String = str
ISO8601 = str
Struct = dict
Float = float
from gaia_sdk.graphql.request.enumeration.RuntimeState import RuntimeState
from gaia_sdk.graphql.request.enumeration.SkillState import SkillState

@dataclass
class BehaviourConnection:
    """
    Represents a paginated collection of behaviours
    """
    dictionary: dict
    @property
    def edges(self) -> List[BehaviourEdge]:
        return list(map(lambda x: BehaviourEdge(x), self.dictionary.get("edges")))
    @property
    def page_info(self) -> PageInfo:
        return PageInfo(self.dictionary.get("pageInfo"))
